sourced/ml/transformers/content2ids.py

```diff
@@ -0,0 +1,146 @@
+import gzip
+import operator
+import os
+import yaml
+
+import pygments
+from pygments.formatter import Formatter
+from pygments.lexers import get_lexer_by_name, ClassNotFound
+from pyspark import Row
+
+from sourced.ml.algorithms import TokenParser
+from sourced.ml.transformers import Transformer
+
+
+class Content2Ids(Transformer):
+
+    class FormatterProxy(Formatter):
+        name = "Proxy"
+        aliases = ["proxy"]
+        filenames = []
+
+        def __init__(self, **options):
+            super(Content2Ids.FormatterProxy, self).__init__(**options)
+            self.callback = options["callback"]
+
+        def format(self, tokensource, outfile):
+            self.callback(tokensource)
+
+    def __init__(self, args, documents_column, **kwargs):
```

Comment:
Transformers should never accept bulk `argparse.Namespace`-s. You need to define the set of args/kwargs and specify them explicitly.

Also would be nice to have type annotations.